kratos/containers/pointer_vector_set.h

```diff
@@ -261,124 +263,253 @@ class PointerVectorSet final
     }
-    ptr_iterator               ptr_begin()
+
+    /**
+     * @brief Returns an iterator pointing to the beginning of the underlying data container.
+     * @return An iterator pointing to the beginning of the underlying data container.
+     */
+    ptr_iterator ptr_begin()
     {
         return mData.begin();
     }
-    ptr_const_iterator         ptr_begin() const
+
+    /**
+     * @brief Returns a constant iterator pointing to the beginning of the underlying data container.
+     * @return A constant iterator pointing to the beginning of the underlying data container.
+     */
+    ptr_const_iterator ptr_begin() const
     {
         return mData.begin();
     }
-    ptr_iterator               ptr_end()
+
+    /**
+     * @brief Returns an iterator pointing to the end of the underlying data container.
+     * @return An iterator pointing to the end of the underlying data container.
+     */
+    ptr_iterator ptr_end()
     {
         return mData.end();
     }
-    ptr_const_iterator         ptr_end() const
+
+    /**
+     * @brief Returns a constant iterator pointing to the end of the underlying data container.
+     * @return A constant iterator pointing to the end of the underlying data container.
+     */
+    ptr_const_iterator ptr_end() const
     {
         return mData.end();
     }
-    ptr_reverse_iterator       ptr_rbegin()
+
+    /**
+     * @brief Returns a reverse iterator pointing to the beginning of the underlying data container.
+     * @return A reverse iterator pointing to the beginning of the underlying data container.
+     */
+    ptr_reverse_iterator ptr_rbegin()
     {
         return mData.rbegin();
     }
+
+    /**
+     * @brief Returns a constant reverse iterator pointing to the beginning of the underlying data container.
+     * @return A constant reverse iterator pointing to the beginning of the underlying data container.
+     */
     ptr_const_reverse_iterator ptr_rbegin() const
     {
         return mData.rbegin();
     }
-    ptr_reverse_iterator       ptr_rend()
+
+    /**
+     * @brief Returns a reverse iterator pointing to the end of the underlying data container.
+     * @return A reverse iterator pointing to the end of the underlying data container.
+     */
+    ptr_reverse_iterator ptr_rend()
     {
         return mData.rend();
     }
+
+    /**
+     * @brief Returns a constant reverse iterator pointing to the end of the underlying data container.
+     * @return A constant reverse iterator pointing to the end of the underlying data container.
+     */
     ptr_const_reverse_iterator ptr_rend() const
     {
         return mData.rend();
     }
 
-    reference        front()       /* nothrow */
+    /**
+     * @brief Returns a reference to the first element in the container.
+     * @note This function is marked as "nothrow," and it asserts that the container is not empty.
+     * @return A reference to the first element in the container.
+     */
+    reference front() /* nothrow */
```

Comment:
why not make these functions `nothrow` for real? It would help the compiler a bit and make your docstring *actually true*.

Reply:
But that would be a behaviour change and I prefer to change it in a independent PR

Reply:
@matekelemen i think it is a good idea. They were originally commented out because some old compiler was not allowing it, but i hope it is now solved

Reply:
Error in compilation: "nothrow’ does not name a type". I will remove it

Reply:
> Error in compilation: "nothrow’ does not name a type".

[noexcept](https://en.cppreference.com/w/cpp/language/noexcept_spec)

Reply:
I think we were expecteding noexcept, not nothorw that doesn't exist. Modifying

Reply:
> > Error in compilation: "nothrow’ does not name a type".
> 
> [noexcept](https://en.cppreference.com/w/cpp/language/noexcept_spec)

Strange, this message appeared after I posted my message, anyway I changed it